support/cas-server-support-consent-rest/src/main/java/org/apereo/cas/consent/RestConsentRepository.java

```diff
@@ -110,4 +112,24 @@ public boolean storeConsentDecision(final ConsentDecision decision) {
         }
         return false;
     }
+    
+    @Override
+    public boolean deleteConsentDecision(final long decisionId, final String principal) {
+        try {
+            final HttpHeaders headers = new HttpHeaders();
+            headers.setAccept(CollectionUtils.wrap(MediaType.APPLICATION_JSON));
+            final Map<String, Object> body = new HashMap<>();
+            body.put("id", decisionId);
+            body.put("principal", principal);
+
+            final HttpEntity<Map> entity = new HttpEntity<>(body, headers);
+            final ResponseEntity<Boolean> result = restTemplate.exchange(this.endpoint, HttpMethod.POST, entity, Boolean.class);
```

Comment:
Perhaps we should do this as the `DELETE` verb of http.

Reply:
I thought about this as well. Since the principal is passed, I decided to use http `POST`. I could drop this and use `DELETE` with only the decision id.
What's your preference?

Reply:
I am not sure the principal is actually needed for a delete op right? All you should have to know before delete the record identifier. I'd rather just pass the id with delete.